logbook-ktor-client/src/main/kotlin/org/zalando/logbook/client/LogbookClient.kt

```diff
@@ -29,40 +29,43 @@ class LogbookClient(
-        override val key: AttributeKey<LogbookClient> = AttributeKey("LogbookFeature")
+    companion object : HttpClientPlugin<Config, LogbookClient> {
+        private val responseProcessingStageKey: AttributeKey<ResponseProcessingStage> =
+            AttributeKey("Logbook.ResponseProcessingStage")
+        override val key: AttributeKey<LogbookClient> = AttributeKey("LogbookPlugin")
         override fun prepare(block: Config.() -> Unit): LogbookClient = LogbookClient(Config().apply(block).logbook)
-        override fun install(feature: LogbookClient, scope: HttpClient) {
+        @OptIn(InternalAPI::class)
+        override fun install(plugin: LogbookClient, scope: HttpClient) {
             scope.sendPipeline.intercept(HttpSendPipeline.Monitoring) {
                 val request = ClientRequest(context)
-                val requestWritingStage = feature.logbook.process(request)
+                val requestWritingStage = plugin.logbook.process(request)
                 val proceedWith = when {
                     request.shouldBuffer() -> {
                         val content = (it as OutgoingContent).readBytes(scope)
                         request.buffer(content)
                         ByteArrayContent(content)
                     }
+
                     else -> it
                 }
                 val responseStage = requestWritingStage.write()
                 context.attributes.put(responseProcessingStageKey, responseStage)
                 proceedWith(proceedWith)
             }
 
-            scope.receivePipeline.intercept(HttpReceivePipeline.After) {
-                val (loggingContent, responseContent) = it.content.split(it)
+            scope.receivePipeline.intercept(HttpReceivePipeline.After) { httpResponse ->
+                val (loggingContent, responseContent) = httpResponse.content.split(httpResponse)
 
-                val responseProcessingStage = it.call.attributes[responseProcessingStageKey]
-                val response = ClientResponse(it)
-                val responseWritingStage = responseProcessingStage.process(response)
-                if (response.shouldBuffer() && !loggingContent.isClosedForRead) {
+                val responseProcessingStage = httpResponse.call.attributes[responseProcessingStageKey]
+                val clientResponse = ClientResponse(httpResponse)
+                val responseWritingStage = responseProcessingStage.process(clientResponse)
+                if (clientResponse.shouldBuffer() && !loggingContent.isClosedForRead) {
                     val content = loggingContent.readBytes()
-                    response.buffer(content)
+                    clientResponse.buffer(content)
                 }
                 responseWritingStage.write()
 
-                val proceedWith = context.wrapWithContent(responseContent).response
+                val proceedWith = httpResponse.call.wrapWithContent(responseContent).response
```

Comment:
This line needs revision.
I just put something here so that the class compiles and the tests pass.

Reply:
I'm also not an expert in Ktor, but this change seems right :thinking: . If you compare where `context.wrapWithContent` was used in 1.6, now this construct is used by the library itself. e.g. in `io.ktor.client.features.observer.ResponseObserver`

Reply:
I'm also not an expert in Ktor, but this change seems right :thinking: . If you compare where `context.wrapWithContent` was used in 1.6, now this construct is used by the library itself. e.g. in `io.ktor.client.features.observer.ResponseObserver`